src/Code/Enemies/Enemy.gd

```diff
@@ -60,17 +60,17 @@ func _on_Hitbox_area_entered(area):
 		if area.has_method("can_damage"):
 			if not area.can_damage(self):
 				return
-		damage(area.damage)
+		damage(area.damage, area.get_class())
 		_on_hit_effects(area)
 	if "knockback" in area:
 		knockback_dir = (global_position - area.global_position).normalized() * area.knockback
 
 
-func damage(amt: float):
+func damage(amt: float, ability_source:String=""):
```

Comment:
Why does the enemy only know about the name of the ability that killed it for this signal? It might be a little more flexible when the enemy dies they refer to the class of the ability through the signal, for things like triggering death effects.